pkg/internal/tests/cross-tests/tfwrite.go

```diff
@@ -16,115 +16,70 @@
 package crosstests
 
 import (
-	"fmt"
-	"io"
-	"sort"
-
-	"github.com/hashicorp/hcl/v2/hclwrite"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
-	"github.com/zclconf/go-cty/cty"
+
+	crosstestsimpl "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/internal/tests/cross-tests/impl"
 )
 
-// Provider writes a provider declaration to an HCL file.
-//
-// Note that unknowns are not yet supported in cty.Value, it will error out if found.
-func writeProvider(out io.Writer, sch map[string]*schema.Schema, typ string, config cty.Value) error {
-	if !config.IsWhollyKnown() {
-		return fmt.Errorf("WriteHCL cannot yet write unknowns")
+func sdkV2NestingToShim(nesting string) crosstestsimpl.Nesting {
+	switch nesting {
+	case "NestingSingle":
+		return crosstestsimpl.NestingSingle
+	case "NestingList":
+		return crosstestsimpl.NestingList
+	case "NestingSet":
+		return crosstestsimpl.NestingSet
+	default:
+		return crosstestsimpl.NestingInvalid
 	}
-	f := hclwrite.NewEmptyFile()
-	block := f.Body().AppendNewBlock("provider", []string{typ})
-	writeBlock(block.Body(), sch, config.AsValueMap())
-	_, err := f.WriteTo(out)
-	return err
 }
 
-// Resource writes a resource declaration to an HCL file.
-//
-// Note that unknowns are not yet supported in cty.Value, it will error out if found.
-func writeResource(
-	out io.Writer, sch map[string]*schema.Schema, resourceType, resourceName string, config cty.Value,
-) error {
-	if !config.IsWhollyKnown() {
-		return fmt.Errorf("WriteHCL cannot yet write unknowns")
-	}
-	f := hclwrite.NewEmptyFile()
-	block := f.Body().AppendNewBlock("resource", []string{resourceType, resourceName})
-	writeBlock(block.Body(), sch, config.AsValueMap())
-	_, err := f.WriteTo(out)
-	return err
+type hclSchemaSDKv2 struct {
+	sch map[string]*schema.Schema
 }
```

Comment:
Style nit: this can just be a type wrapper. It doesn't need to be an enclosing struct.
```suggestion
type hclSchemaSDKv2 map[string]*schema.Schema
```